### ISSUE TITLE
Make env variable caching and lazy loading of packages optional 

### ISSUE THREAD

**Issue by robbyoconnor** [syl20bnr/spacemacs#10864]
@syl20bnr -- I love spacemacs and I'm very unlikely to ever leave it...but the latest commits to `develop` are a bit of a headache for me right now. I get what you're trying to do but some of this is not necessary. We need to have tests to check that you didn't introduce regressions. 

That said, with all the weird bugs and problems that lazy loading the env variable and lazy loading of packages is causing, can we **_PLEASE_** make it optional...for those like me who don't seem to have an issue...

**Comment by jbharat** [comment]
I run emacs as a daemon and connect to it with emacsclient. It was fast for me even before the latest changes. It would be good if there was a dotspacemacs variable to toggle this as suggested by @robbyoconnor

**Comment by fiveNinePlusR** [comment]
It's ok. we are working on something that will work and be fast at startup as well. develop will have regressions unless the test suite is much much more comprehensive and even then emacs is just simply weird and platforms/configs are different and hard to test for in general. It doesn't help that emacs packages are not locked to a specific revision in the main package store. 

> we would still like visibility and ability to test on a branch outside of develop prior to introducing a feature to prevent (not eliminate) potential regressions

that sounds like something worthwhile. not sure it would have enough testers to make it actually a decent firewall for regressions but it might be worth looking into. have a "currently working on" branch that gets merged every week or so into develop?

**Comment by robbyoconnor** [comment]
maybe make a branch called `canary`

**Comment by ljupchokotev** [comment]
A `canary` branch sounds nice. Even if we don't have a lot of testers, it will still catch bugs like the current ones that happened with the env-vars cache and lazy loading.

Another way would be to have more frequent releases on `master`, so the users don't need to use `develop` to have the latest and greatest in Spacemacs.

**Comment by syl20bnr** [comment]
Creating a new branch is not the right thing to do, it just replicates
poorly what we already have.

Actually develop is even pretty much stable.

It happens that it breaks... rarely, in such case we work on it fast.
The issue with env. var will be fixed this week, this is fast.

In the meantime Spacemacs is distributed via a Git repository where
anyone can consult the reflog and reset their version to something that
fits.

I would prefer that someone asks for or proposes a PR to revert a bad
commit than discussing hours and hours long about the develop branch
that is being broken for some users once every 6 months ;-)

**Comment by robbyoconnor** [comment]
I don't think I've had `develop` break on me for awhile...I can't even remember when it broke for me last...

**Comment by NerdsvilleCEO** [comment]
Agreed, I think making the option for caching or not would be awesome 💜💜💜

**Comment by syl20bnr** [comment]
Actually closing as duplication of https://github.com/syl20bnr/spacemacs/issues/10851
Thanks for your patience.

**Comment by cviebig** [comment]
Sorry for waking up that one here, but actually I noticed a year ago that lazy loading did not work well with batch execution of emacs from the shell, because the command is just executed before loading actually finished. For me that's the main reason to favor an optional non-lazy-loading startup too. It's not my main priority right now, however as I just found this issue, I wanted to add this. So if anyone is having the same issue and did find this ticket, feel free to create a proper report to follow-up on that.
